### ISSUE TITLE
`Instant#toLocaleString` defaults to UTC instead of local time zone

### ISSUE THREAD

**Issue by stesen-wefunder** [fullcalendar/temporal-polyfill#12]
According to the [official Temporal documentation](https://tc39.es/proposal-temporal/docs/instant.html#toLocaleString) (emphasis mine):
> Because Temporal.Instant does not carry a time zone, the time zone used for the output will be the timeZone property of options, if present; **and otherwise, the current time zone from the environment, which is usually the system's time zone.**

However, in this implementation calling `toLocaleString()` on an `Instant` with no arguments seems to default to UTC.

It seems to be happening on this line:
https://github.com/fullcalendar/temporal/blob/ed6586020b157cf9ad647e2688dd869542c895f2/packages/temporal-polyfill/src/native/intlFactory.ts#L188

Based on my own manual testing with the `Intl.DateTimeFormat` API, as well as the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat), simply removing the default and letting the time zone ID be undefined should provide this behavior automatically:
```js
new Intl.DateTimeFormat(undefined, {
    year: "numeric",
    month: "numeric",
    day: "numeric",
    weekday: undefined,
    hour: "numeric",
    minute: "2-digit",
    second: "2-digit",
    timeZoneName: "long"
}).format(Date.now())
// => "5/19/2022, 6:02:01 PM Pacific Daylight Time"
```

**Comment by arshaw** [comment]
Confirmed, this is a bug. I'll fix in the next round of code changes: https://github.com/fullcalendar/temporal#project-update-2023-04-22

**Comment by arshaw** [comment]
This has been fixed in [v0.2.0](https://github.com/fullcalendar/temporal-polyfill/releases/tag/v0.2.0).

Please let me know once you confirm. Thanks!
